decoder/expr_any_semtok.go

```diff
@@ -186,3 +188,34 @@ func (a Any) semanticTokensForOperatorExpr(ctx context.Context) ([]lang.Semantic
 
 	return tokens, false
 }
+
+func (a Any) semanticTokensForTemplateExpr(ctx context.Context) ([]lang.SemanticToken, bool) {
+	tokens := make([]lang.SemanticToken, 0)
+
+	switch eType := a.expr.(type) {
+	case *hclsyntax.TemplateExpr:
+		if eType.IsStringLiteral() {
+			return nil, false
+		}
+
+		for _, partExpr := range eType.Parts {
+			cons := schema.AnyExpression{
+				OfType: cty.String,
+			}
+			expr := newExpression(a.pathCtx, partExpr, cons)
+			tokens = append(tokens, expr.SemanticTokens(ctx)...)
+		}
+
+		return tokens, true
```

Comment:
I'm not sure why but it looks like that highlighting of HEREDOC still needs accounting for.

![Screenshot 2023-11-21 at 13 09 43](https://github.com/hashicorp/hcl-lang/assets/287584/572d7fc5-cc61-46ee-8014-eba1886d032b)